src/hooks/usePermitAllowance.ts

```diff
@@ -0,0 +1,88 @@
+import {
+  AllowanceData,
+  AllowanceProvider,
+  AllowanceTransfer,
+  MaxAllowanceTransferAmount,
+  PERMIT2_ADDRESS,
+  PermitSingle,
+} from '@uniswap/permit2-sdk'
+import { Token } from '@uniswap/sdk-core'
+import { useWeb3React } from '@web3-react/core'
+import ms from 'ms.macro'
+import { useCallback, useEffect, useMemo, useState } from 'react'
+
+const PERMIT_EXPIRATION = ms`30d`
+const PERMIT_SIG_EXPIRATION = ms`30m`
+
+function toDeadline(expiration: number): number {
+  return Math.floor((Date.now() + expiration) / 1000)
+}
+
+export function usePermitAllowance(token?: Token, spender?: string) {
+  const { account, provider } = useWeb3React()
+  const allowanceProvider = useMemo(() => provider && new AllowanceProvider(provider, PERMIT2_ADDRESS), [provider])
```

Comment:
Why is there a new provider used here? I expected the PERMIT2_ADDRESS to be used with a contract constructor instead.

Reply:
This is just a helper provider from permit2-sdk that wraps the contract object with nicer helpers